modules/base/rendering/renderabletrailorbit.cpp

```diff
@@ -0,0 +1,450 @@
+            }
+        }
+    };
+
+    // Insert the parents documentation entries until we have a verifier that can deal
+    // with class hierarchy
+    openspace::Documentation parentDoc = RenderableTrail::Documentation();
+    doc.entries.insert(
+        doc.entries.end(),
+        parentDoc.entries.begin(),
+        parentDoc.entries.end()
+    );
+
+    return doc;
+}
+
+RenderableTrailOrbit::RenderableTrailOrbit(const ghoul::Dictionary& dictionary)
+    : RenderableTrail(dictionary)
+    , _period("period", "Period in days", 0.0, 0.0, 1e9)
+    , _resolution("resoluion", "Number of Samples along Orbit", 10000, 1, 1e6)
+    , _needsFullSweep(true)
+    , _indexBufferDirty(true)
+{
+    documentation::testSpecificationAndThrow(
+        Documentation(),
+        dictionary,
+        "RenderableTrailOrbit"
+    );
+
+    // Period is in days
+    using namespace std::chrono;
+    int factor = duration_cast<seconds>(hours(24)).count();
+    _period = dictionary.value<double>(KeyPeriod) * factor;
+    _period.onChange([&] { _needsFullSweep = true; _indexBufferDirty = true; });
+    addProperty(_period);
+
+    _resolution = static_cast<int>(dictionary.value<double>(KeyResolution));
+    _resolution.onChange([&] { _needsFullSweep = true; _indexBufferDirty = true; });
+    addProperty(_resolution);
+
+    // We store the vertices with (excluding the wrapping) decending temporal order
+    _primaryRenderInformation.sorting = RenderInformation::VertexSorting::NewestFirst;
+}
+
+bool RenderableTrailOrbit::initialize() {
+    bool res = RenderableTrail::initialize();
+
+    glGenVertexArrays(1, &_primaryRenderInformation._vaoID);
+    glGenBuffers(1, &_primaryRenderInformation._vBufferID);
+    glGenBuffers(1, &_primaryRenderInformation._iBufferID);
+
+    return res;
+}
+
+bool RenderableTrailOrbit::deinitialize() {
+    glDeleteVertexArrays(1, &_primaryRenderInformation._vaoID);
+    glDeleteBuffers(1, &_primaryRenderInformation._vBufferID);
+    glDeleteBuffers(1, &_primaryRenderInformation._iBufferID);
+
+    return RenderableTrail::deinitialize();
+}
+
+void RenderableTrailOrbit::update(const UpdateData& data) {
+    // Overview:
+    // 1. Update trails
+    // 2. Update floating position
+    // 3. Determine which parts of the array to upload and upload the data
+
+
+    // 1
+    // Update the trails; the report contains whether any of the other values has been
+    // touched and if so, how many
+    UpdateReport report = updateTrails(data);
+
+    // 2
+    // Write the current location into the floating position
+    glm::vec3 p = _translation->position(data.time);
+    _vertexArray[_primaryRenderInformation.first] = { p.x, p.y, p.z };
+
+    glBindVertexArray(_primaryRenderInformation._vaoID);
+    glBindBuffer(GL_ARRAY_BUFFER, _primaryRenderInformation._vBufferID);
+
+    // 3
+    if (!report.needsUpdate) {
+        // If no other values have been touched, we only need to upload the floating value
+        glBufferSubData(
```

Comment:
Added a boolean to the UpdateData structure and test for both 0 delta time and pausing